app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,57 @@
+class PasswordResetsController < ApplicationController
+  PERMITTED_PARAMS = %i(password password_confirmation).freeze
+  before_action :get_user, :valid_user, :check_expiration, only: %i(edit update)
+
+  def new; end
+
+  def create
+    @user = User.find_by email: params[:password_reset][:email].downcase
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t "flash.pwd_reset.info"
+      redirect_to root_url
+    else
+      flash.now[:danger] = t "flash.pwd_reset.danger.not_found"
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].empty?
```

Comment:
```suggestion
    if user_params[:password].empty?
```
Tìm hiểu empty, blank xem nên dùng cái nào